java/core/src/main/java/sleeper/core/util/PollWithRetries.java

```diff
@@ -26,6 +26,10 @@
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
+/**
+ * A class to assist with periodic polling and retrying. When started, this class will periodically poll until
+ * either an exit condition has been met or a maximum number of polls has been reached
```

Comment:
This should have a full stop at the end.

The "A class" and "this class" is probably not needed. The point is to poll until a condition is met.